sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/binaryfile/BinaryFileFormat.scala

```diff
@@ -206,6 +188,23 @@ object BinaryFileFormat {
       case _ => (_ => true)
     }
   }
+
+  private[binaryfile] def genPrunedRow(
+      path: String,
+      status: FileStatus,
+      readContent: => Array[Byte],
+      requiredFieldNames: Array[String]): InternalRow = {
+
+    val values = requiredFieldNames.map {
+      case PATH => UTF8String.fromString(path)
+      case LENGTH => status.getLen
+      case MODIFICATION_TIME => DateTimeUtils.fromMillis(status.getModificationTime)
+      case CONTENT => readContent
+      case name => throw new RuntimeException(s"Unexcepted field name: ${name}")
+    }
+    InternalRow(values: _*)
```

Comment:
Do we need to change `inferSchema()` or still return `content` field with null values? cc: @cloud-fan

Reply:
What about adding a "keep invalid" option, when file read error, fill content column "null"?
Now when file loaded error, the datasource loading broken.